### ISSUE TITLE
Some questions about the loss function

### ISSUE THREAD

**Issue by cici-97** [zzs1994/D-LADMM#2]
Thank you so much for sharing! I've been working on the paper _Differentiable Linearized ADMM_ and there are some questions about the training process without the ground truth.

In this paper, the loss function in this case consists of the original objective function (1) and the dual function of (1). But in the code, I could not find the corresponding code of the dual function **(d(λK))** accurately. 
According to the term **"torch.mean(L[k] * input_bs_var)"** , I guess that the part 
**"torch.mean(dual_gap(torch.mm(A_tensor.t(), L[k]), alpha)) + torch.mean(dual_gap(L[k], 1)) + torch.mean(L[k] * input_bs_var))"** 
should be the code corresponding to the dual equation **(d(λK))**, but I can't understand why the **dual_gap** function is used, which is softplus in essence, and **Z** and **E** are lost here.

Hope you can help me answer!

**Comment by XingyuXie** [comment]
@cici-97 
Thx for your question.
The dual function of the loss we used in this experiment is a step function in closed form, which can easily be checked. Hence, we utilize its smooth version as an alternative to ensure training efficiency and stability. That's why the function softplus appears here.

For more discussion, do not hesitate to email us. Thx again.

**Comment by cici-97** [comment]
@XingyuXie 
Thank you very much for your prompt reply! Following your hints, I have rederived the formula and understood the code. Your work is very interesting and I love it!

**Comment by XingyuXie** [comment]
@cici-97 
Thank you for your interest in our work. I close this issue. If you have any more questions, please contact us. My WeChat: xyxie_joy
